### PR TITLE
Fix numpy block linebreaks

### PR DESCRIPTION
**Pull request by machow**
This PR addresses #167. I'm happy to tweak / look into supporting different cases. (I've xfailed the test of what would be a breaking change, and added a test of the new supported behavior.)

### CODE REVIEW

File: src/griffe/docstrings/numpy.py [hunk 1]
```diff
@@ -648,6 +665,7 @@ def parse(
     *,
     ignore_init_summary: bool = False,
     trim_doctest_flags: bool = True,
+    allow_block_breaks: bool = False,
```
Comment (pawamoy) [inline src/griffe/docstrings/numpy.py:668]: Shouldn't it be true to be prevent a breaking change? By the way I'm confused by the parameter name and description. I keep going back to the logic above to understand what it does. I think naming it `allow_section_blank_line` would be better :thinking: If true, then one needs *two* blank lines to break a section. WDYT?

Comment (machow) [reply src/griffe/docstrings/numpy.py:668]: Hey, sorry for the wait--I was out of town last week. Agreed the parameter name is confusing, lemme go ahead and switch it to `allow_section_blank_line` and fix the default.

Comment (pawamoy) [reply src/griffe/docstrings/numpy.py]: No problem! Lets not be sorry for life things ^^ Thanks for the follow-up :)

**Comment by pawamoy** [comment]
Ugh, I included tests for inheritance in the main branch, but the feature is not merged yet... let me fix CI.

**Review by pawamoy — APPROVED** [review]
LGTM

**Comment by pawamoy** [comment]
Thanks @machow!

**Pull request merged**